cot/src/response/into_response.rs

```diff
@@ -0,0 +1,258 @@
+
+impl<R, E> IntoResponse for Result<R, E>
+where
+    R: IntoResponse,
+    E: IntoResponse,
+{
+    fn into_response(self) -> Response {
+        match self {
+            Ok(value) => value.into_response(),
+            Err(err) => err.into_response(),
+        }
+    }
+}
+
+impl IntoResponse for Response {
+    fn into_response(self) -> Response {
+        self
+    }
+}
+
+// Text implementations
+
+impl_into_response_for_type_and_mime!(&'static str, mime::TEXT_PLAIN_UTF_8);
+impl_into_response_for_type_and_mime!(String, mime::TEXT_PLAIN_UTF_8);
+
+impl IntoResponse for Box<str> {
+    fn into_response(self) -> Response {
+        String::from(self).into_response()
+    }
+}
+
+// Bytes implementations
+
+impl_into_response_for_type_and_mime!(&'static [u8], mime::APPLICATION_OCTET_STREAM);
+impl_into_response_for_type_and_mime!(Vec<u8>, mime::APPLICATION_OCTET_STREAM);
+impl_into_response_for_type_and_mime!(Bytes, mime::APPLICATION_OCTET_STREAM);
+
+impl<const N: usize> IntoResponse for &'static [u8; N] {
+    fn into_response(self) -> Response {
+        self.as_slice().into_response()
+    }
+}
+
+impl<const N: usize> IntoResponse for [u8; N] {
+    fn into_response(self) -> Response {
+        self.to_vec().into_response()
+    }
+}
+
+impl IntoResponse for Box<[u8]> {
+    fn into_response(self) -> Response {
+        Vec::from(self).into_response()
+    }
+}
+
+impl IntoResponse for BytesMut {
+    fn into_response(self) -> Response {
+        self.freeze().into_response()
+    }
+}
+
+// HTTP structures for common uses
+
+impl IntoResponse for StatusCode {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.status_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::HeaderMap {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.headers_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::Extensions {
+    fn into_response(self) -> Response {
+        let mut res = ().into_response();
+        *res.extensions_mut() = self;
+        res
+    }
+}
+
+impl IntoResponse for http::response::Parts {
+    fn into_response(self) -> Response {
+        Response::from_parts(self, Body::empty())
+    }
+}
+
+// Data type structures implementations
+
+impl IntoResponse for Html {
+    /// Create a new HTML response.
+    ///
+    /// This creates a new [`Response`] object with a content type of
+    /// `text/html; charset=utf-8` and given status code and body.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::html::Html;
+    /// use cot::response::IntoResponse;
+    ///
+    /// let html = Html::new("<div>Hello</div>");
+    ///
+    /// let response = html.into_response();
+    /// ```
+    fn into_response(self) -> Response {
+        self.as_str().to_owned().into_response().with_content_type(
+            http::header::HeaderValue::from_static(mime::TEXT_HTML_UTF_8.as_ref()),
+        )
+    }
+}
+
```

Comment:
An implementation for the `Json` struct should is very much needed, too. I'm wondering if it should be extracted somewhere outside `extractors.rs`.

Reply:
In general it should be fairly easy, but first we need to figure out the best solution for using the same struct for both path extractor and user-created response (take a look [here](https://github.com/cot-rs/cot/pull/256#discussion_r2062576402)). Because of that, I'd leave it for a separate PR following the merging of this one.

Reply:
https://github.com/cot-rs/cot/pull/256#discussion_r2065623335